templates/qaoa.yaml

```diff
@@ -17,19 +21,83 @@ spec:
             from zquantum.qaoa.maxcut import get_maxcut_hamiltonian
             from qeopenfermion import save_qubit_operator
             graph = load_graph('graph.json')
-            hamiltonian = get_maxcut_hamiltonian(graph)
+            hamiltonian = get_maxcut_hamiltonian(graph, scaling={{inputs.parameters.scaling}}, shifted={{inputs.parameters.shifted}})
             save_qubit_operator(hamiltonian,'hamiltonian.json')
     outputs:
       artifacts:
       - name: hamiltonian
         path: /app/hamiltonian.json
 
-  - name: solve-maxcut-by-exhaustive-search
+
+  - name: generate-list-maxcut-hamiltonians
     parent: generic-task
     inputs:
       parameters:
       - name: command
         value: python3 main_script.py
+      - name: graph-specs
+      - name: number-of-instances
+      - name: number-of-qubits
+      - name: shifted
+        default: "False"
+      - name: scaling
+        value: "1.0"       
+      artifacts:
+      - name: main-script
+        path: /app/main_script.py
+        raw:
+          data: |
+            from zquantum.qaoa.maxcut import get_random_maxcut_hamiltonians
+            from qeopenfermion import save_qubit_operator_set
+            hamiltonians = get_random_maxcut_hamiltonians(
+                                    {{inputs.parameters.graph-specs}},
+                                    {{inputs.parameters.number-of-instances}},
+                                    {{inputs.parameters.number-of-qubits}},
+                                    shifted = {{inputs.parameters.shifted}},
+                                    scaling = {{inputs.parameters.scaling}}
+                                )
+            save_qubit_operator_set(hamiltonians, 'hamiltonians.json')
+    outputs:
+      artifacts:
+      - name: hamiltonians
+        path: /app/hamiltonians.json
+        
+  - name: generate-list-ansatzes-from-hamiltonians
```

Comment:
This name doesn't reflect reality really.
I'd change it to "create-circuits-from-hamiltonians-for-farhi-ansatz" or sth like this.

Reply:
8519100d1412e066c2b1a38348798b0f26aef792